### PR TITLE
IGNITE-15149 Javadoc fixes for configuration-api module.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-15149

### CODE REVIEW

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java [hunk 1]
```diff
@@ -19,7 +19,11 @@
 
 import java.util.function.Consumer;
 
-/** */
+/**
+ * Change closure parameter for the {@link NamedConfigurationTree} instance. Contains methods to modify named lists.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java:23]: "Change closure parameter" - sound very ambiguous, can you rephrase it please?

Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java:23]:
```suggestion
 * Change closure parameter for a {@link NamedConfigurationTree} instance. Contains methods to modify named lists.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java [hunk 2]
```diff
@@ -19,7 +19,11 @@
...
+ *
+ * @param <Change> Change type for the named list elements.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java:25]: What is a "Change type"?

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java [hunk 3]
```diff
@@ -19,9 +19,15 @@
 
 import java.util.List;
 
-/** */
-public interface NamedListView<T> {
+/**
+ * View type for the {@link NamedConfigurationTree}. Represents immutable snapshot of the named list configuration.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java:23]:
```suggestion
 * View type for a {@link NamedConfigurationTree}. Represents an immutable snapshot of a named list configuration.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java (hunk 2 above)
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java:25]:
```suggestion
 * @param <Change> Change type for named list elements.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java [hunk 4]
```diff
@@ -19,9 +19,15 @@
...
+ *
+ * @param <View> View type for the named list element.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java:25]: What is a "View type"?

Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java:25]:
```suggestion
 * @param <View> View type for named list elements.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java [hunk 5]
```diff
@@ -19,9 +19,15 @@
...
+ */
+public interface NamedListView<View> {
     /**
+     * Returns immutable collection of keys contained within this list.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java:29]:
```suggestion
     * Returns an immutable collection of keys contained within this list.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java [hunk 6]
```diff
@@ -32,9 +38,11 @@
      * @param key Key string.
      * @return Requested value or {@code null} if it's not found.
      */
-    T get(String key);
+    View get(String key);
 
     /**
+     * Returns number of elements.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListView.java:44]:
```suggestion
     * Returns the number of elements in this list.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java [hunk 7]
```diff
@@ -46,20 +59,26 @@ public RootKey(Class<?> schemaClass) {
     }
 
     /**
+     * Returns name of the configuration tree.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java:62]:
```suggestion
     * Returns the name of the configuration root.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java [hunk 8]
```diff
@@ -46,20 +59,26 @@ public RootKey(Class<?> schemaClass) {
...
+     *
      * @return Name of the configuration root.
      */
     public String key() {
         return rootName;
     }
 
     /**
+     * Returns configuration type of the root.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java:71]:
```suggestion
     * Returns the configuration type of the root.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java [hunk 9]
```diff
@@ -46,20 +59,26 @@ public RootKey(Class<?> schemaClass) {
...
+     *
      * @return Configuration type of the root.
      */
     public ConfigurationType type() {
         return storageType;
     }
 
     /**
+     * Returns schema class for the root.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java:80]:
```suggestion
     * Returns the schema class for the root.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/annotation/ConfigurationRoot.java [hunk 10]
```diff
@@ -34,9 +34,17 @@
 @Retention(RUNTIME)
 @Documented
 public @interface ConfigurationRoot {
-    /** @return Unique root name. */
+    /**
+     * Returns unique root name.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/annotation/ConfigurationRoot.java:38]: same stuff about articles

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/notifications/ConfigurationNotificationEvent.java [hunk 11]
```diff
@@ -23,24 +23,31 @@
...
  * Event object propogated on configuration change. Passed to listeners after configuration changes are applied.
  *
+ * @param <VIEW> Type of the subtree or the value that has been changed.
  * @see ConfigurationProperty#listen(ConfigurationListener)
  * @see ConfigurationListener
  * @see ConfigurationNotificationEvent
  */
 public interface ConfigurationNotificationEvent<VIEW> {
     /**
+     * Returns previous value of the updated configuration.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/notifications/ConfigurationNotificationEvent.java:33]: same stuff about articles

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/Max.java [hunk 12]
```diff
@@ -31,5 +31,10 @@
 @Retention(RUNTIME)
 @Documented
 public @interface Max {
+    /**
+     * Returns upped bound for the value.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/Max.java:35]:
```suggestion
     * Returns the upper bound for the value.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/Min.java [hunk 13]
```diff
@@ -31,5 +31,10 @@
 @Retention(RUNTIME)
 @Documented
 public @interface Min {
+    /**
+     * Returns lower bound for the value.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/Min.java:35]:
```suggestion
     * Returns the lower bound for the value.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/ValidationContext.java [hunk 14]
```diff
@@ -25,19 +25,34 @@
 /**
  * Validation context for the validators.
  *
+ * @param <VIEW> Type of the subtree or the value that is being validated.
  * @see Validator#validate(Annotation, ValidationContext)
  */
 public interface ValidationContext<VIEW> {
-    /** @return String representation of currently validated value, i.e. {@code root.config.node} */
+    /**
+     * Returns key of validated node.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/ValidationContext.java:33]: same

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/ValidationIssue.java [hunk 15]
```diff
@@ -18,19 +18,25 @@
 
 import org.apache.ignite.internal.tostring.S;
 
-/** */
+/**
+ * Class that stores information about issues found during configuration valudation.
```
Comment (sashapolo) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/ValidationIssue.java:22]:
```suggestion
 * Class that stores information about issues found during the configuration validation.
```

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java (hunk 1 above)
Comment (ibessonov) [reply modules/configuration-api/src/main/java/org/apache/ignite/configuration/NamedListChange.java:23]: Sure

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/annotation/ConfigurationRoot.java (hunk 10 above)
Comment (ibessonov) [reply modules/configuration-api/src/main/java/org/apache/ignite/configuration/annotation/ConfigurationRoot.java]: I hope I fixed it right

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/notifications/ConfigurationNotificationEvent.java (hunk 11 above)
Comment (ibessonov) [reply modules/configuration-api/src/main/java/org/apache/ignite/configuration/notifications/ConfigurationNotificationEvent.java]: Done

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/ValidationContext.java (hunk 14 above)
Comment (ibessonov) [reply modules/configuration-api/src/main/java/org/apache/ignite/configuration/validation/ValidationContext.java]: Done

**Review by sashapolo — APPROVED** [review]
